src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Claims/WindowsClaimSet.cs

```diff
@@ -189,13 +256,13 @@ public override IEnumerable<Claim> FindClaims(string claimType, string right)
                 {
                     if (right == null || Rights.Identity == right)
                     {
-                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity.User, Rights.Identity);
+                        yield return new Claim(ClaimTypes.Sid, _windowsIdentity, Rights.Identity);
                     }
                 }
 
                 if (right == null || Rights.PossessProperty == right)
                 {
-                    if (TryCreateWindowsSidClaim(_windowsIdentity, out Claim sid))
+                    if (_windowsIdentity is WindowsIdentity && TryCreateWindowsSidClaim((WindowsIdentity)_windowsIdentity, out Claim sid))
                     {
```

Comment:
If you make SupportedClaimType return false when it's not a WindowsIdentity, then you don't need the is check here.